change-validation/src/main/scala/bootstrap/rudder/plugin/ChangeValidationConf.scala

```diff
@@ -131,16 +132,36 @@ class ChangeValidationWorkflowLevelService(
   }
 
   /**
-   * Methode to use to combine several validationNeeded check. It's the same for all objects?
+   * Methode to use to combine several validationNeeded check.
+   * Note that a validated user will prevent workflow to be performed, no other validationNeeded check will be executed
    */
   def combine[T](checkFn:(ValidationNeeded, EventActor, T) => Box[Boolean], checks: Seq[ValidationNeeded], actor: EventActor, change: T): Box[WorkflowService] = {
-    getWorkflow(validationNeeded.foldLeft(Full(false): Box[Boolean]) { case (shouldValidate, nextCheck) =>
-      shouldValidate.flatMap {
-        // logic is "or": if previous should validate is true, don't check following
-        case true  => Full(true)
-        case false => checkFn(nextCheck, actor, change)
-      }
-    })
+    def getWorkflowAux = {
+      getWorkflow(validationNeeded.foldLeft(Full(false): Box[Boolean]) { case (shouldValidate, nextCheck) =>
+        shouldValidate.flatMap {
+          // logic is "or": if previous should validate is true, don't check following
+          case true  => Full(true)
+          case false => checkFn(nextCheck, actor, change)
+        }
+      })
+    }
+
+    /*
+     * Here we check if there is a validated user that should not be subject to any validation workflow
+     * if there is no validated user, we iterate over `checks: Seq[ValidationNeeded]` to verify if there
+     * there is a specific workflow.
+     *
+     * Check why we decided to separate the validated user logic from `ValidationNeeded` objects :
+     * https://issues.rudder.io/issues/22188#note-5
+     */
+    validatedUserRepo.get(actor) match {
+      case Full(ea) =>
```

Comment:
You should at least log the error in the emptybox case and i do even think it should be an error

Also you can do both match in one go

validatedUserRepo.get(actor) match {
  case Full(Some(_)
  case Full(None)
  case eb : Emptybox => eb  (i think it should be an error (or ar least log here then getWorkflowAux, but I think it's an error case)